### PR TITLE
Fix build on Emscripten tip of tree

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

**Review by tlively — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -447,6 +448,7 @@ if(EMSCRIPTEN)
   else()
     target_link_libraries(binaryen_js "-sEXPORT_ES6=1")
   endif()
+  target_link_libraries(binaryen_js "-sDEFAULT_LIBRARY_FUNCS_TO_INCLUDE=[$allocateUTF8OnStack]")
```
Comment (sbc100) [inline CMakeLists.txt:451]: you don't need the square brackets here or above.

Comment (kripken) [reply CMakeLists.txt:451]: I tried without the square brackets, and it hits an escaping issue with the `$`.

Comment (kripken) [reply CMakeLists.txt:451]: Oh, actually my local testing must have been wrong... it's still failing on CI due to escaping :cry:

Comment (kripken) [reply CMakeLists.txt]: I'm trying `$$` now for the escaping. I have to try on CI as for some reason my local CMake behaves differently.

Is there a simpler way to fix the escaping that I'm missing perhaps?

Comment (kripken) [reply CMakeLists.txt]: Looks like I've hit this problem before https://github.com/WebAssembly/binaryen/pull/4793#issuecomment-1180671654

Comment (sbc100) [reply CMakeLists.txt]: You can also use `-sEXPORTED_RUNTIME_METHODS=allocateUTF8OnStack` which doesn't use $-prefixing.

Comment (sbc100) [reply CMakeLists.txt]: Oh I see that you did that :)

**Pull request merged**